cli/commands/pods.go

```diff
@@ -56,55 +70,157 @@ func toPodTree(podsJSONBytes []byte) string {
+	}
+
+	buf.WriteString(fmt.Sprintf("%s%s\n", prefix, pod["name"]))
+
+	rawPodInstances, ok := pod["instances"].([]interface{})
+	if !ok {
+		return
+	}
+	for i, rawPodInstance := range rawPodInstances {
+		appendServicePodInstance(buf, rawPodInstance, lastPod, i == len(rawPodInstances)-1)
+	}
+}
+
+func appendServicePodInstance(buf *bytes.Buffer, rawPodInstance interface{}, lastPod bool, lastPodInstance bool) {
+	podInstance, ok := rawPodInstance.(map[string]interface{})
+	if !ok {
+		return
+	}
+
+	var prefix string
 	if lastPod {
-		podPrefix = "└─ "
+		if lastPodInstance {
+			prefix = "   └─ "
+		} else {
+			prefix = "   ├─ "
+		}
 	} else {
-		podPrefix = "├─ "
+		if lastPodInstance {
+			prefix = "│  └─ "
+		} else {
+			prefix = "│  ├─ "
+		}
 	}
 
-	buf.WriteString(fmt.Sprintf("%s%s\n", podPrefix, podName))
+	buf.WriteString(fmt.Sprintf("%s%s\n", prefix, podInstance["name"]))
 
-	tasks, ok := rawPod.([]interface{})
+	tasks, ok := podInstance["tasks"].([]interface{})
 	if !ok {
 		return
 	}
 	for i, rawTask := range tasks {
-		appendTask(buf, rawTask, lastPod, i == len(tasks)-1)
+		appendServiceTask(buf, rawTask, lastPod, lastPodInstance, i == len(tasks)-1)
 	}
 }
 
-func appendTask(buf *bytes.Buffer, rawTask interface{}, lastPod bool, lastTask bool) {
-	var taskPrefix string
+func appendServiceTask(buf *bytes.Buffer, rawTask interface{}, lastPod bool, lastPodInstance bool, lastTask bool) {
+	task, ok := rawTask.(map[string]interface{})
+	if !ok {
+		return
+	}
+
+	var prefix string
 	if lastPod {
-		if lastTask {
-			taskPrefix = "   └─ "
+		if lastPodInstance {
+			if lastTask {
+				prefix = "      └─ "
+			} else {
+				prefix = "      ├─ "
+			}
 		} else {
-			taskPrefix = "   ├─ "
+			if lastTask {
+				prefix = "   │  └─ "
+			} else {
+				prefix = "   │  ├─ "
+			}
 		}
 	} else {
-		if lastTask {
-			taskPrefix = "│  └─ "
+		if lastPodInstance {
+			if lastTask {
+				prefix = "│     └─ "
+			} else {
+				prefix = "│     ├─ "
+			}
 		} else {
-			taskPrefix = "│  ├─ "
+			if lastTask {
+				prefix = "│  │  └─ "
+			} else {
+				prefix = "│  │  ├─ "
+			}
 		}
 	}
 
+	taskName, ok := task["name"]
+	if !ok {
+		taskName = "<UNKNOWN>"
+	}
+	taskState, ok := task["status"]
+	if !ok {
+		taskState = "<UNKNOWN>"
+	}
+	buf.WriteString(fmt.Sprintf("%s%s (%s)\n", prefix, taskName, taskState))
+}
+
+func toPodTree(podsJSONBytes []byte) string {
+	response, err := client.UnmarshalJSON(podsJSONBytes)
+	if err != nil {
+		client.PrintMessageAndExit(fmt.Sprintf("Failed to parse JSON in pod response: %s", err))
+	}
+
+	var buf bytes.Buffer
+	buf.WriteString(fmt.Sprintf("%s\n", response["name"]))
+
+	i := 0
+	rawTasks, ok := response["tasks"].([]interface{})
+	if !ok {
+		client.PrintMessageAndExit("Failed to parse JSON pods in response")
+	}
+	for _, rawPod := range rawTasks {
+		appendPodTask(&buf, rawPod, i == len(rawTasks)-1)
+		i++
+	}
+
+	// Trim extra newline from end:
+	buf.Truncate(buf.Len() - 1)
+
+	return buf.String()
```

Comment:
Instead of manipulating the buffer directly, could we call a trim-right method on the string representation? The trimming *assumes* that we have a trailing newline, and this may be easy to miss in future refactorings.

Reply:
Switched to `strings.TrimRight()`, a bit more succinct and arguably a bit easier to understand